src/web-component/ay-accordion-header/index.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Ayogo Health Inc
+
+/**
+ * ay-accordion-header acts as a button for its first parent ay-accordion element thereby enabling the toggle functionality
+ *
+ * For example:
+ * ```
+ * <ay-accordion-header>
+ *  Button Name
+ * </ay-accordion-header>
+ * ```
+ *
+ * @name ay-accordion-header
+ *
+ */
+
+const accordionHeaderClickMap : WeakMap<HTMLElement, () => void> = new WeakMap();
+const accordionHeaderPressMap : WeakMap<HTMLElement, (event: KeyboardEvent) => void> = new WeakMap();
+
+export class AyAccordionHeader extends HTMLElement {
+
+  connectedCallback() {
+
+    this.setAttribute('role', 'button');
+    this.setAttribute('tabIndex', '0');
+
+    if(this.closest('ay-accordion').hasAttribute('disabled')) {
```

Comment:
It might be easier to have the accordion find its child header and set the aria-disabled attribute. The way you're doing it right now won't dynamically add/remove the attribute if the accordion is enabled or disabled dynamically.